File: backend/app/src/models/persons/Worker.py
from .Human import Human


class Worker(Human):
    def __init__(self, name: str, employed: bool = False, field_of_work = "unskilled", working: bool = False, production: int = 0):
        super().__init__(name=name, profession="Worker")
        self.employed: bool = employed
        self.field_of_work: str = field_of_work
        self.working: bool = working
        self.production: int = production







if __name__ == "__main__":
    pass

File: frontend/static/script/models/population/Worker.js
"use strict";
import { Human } from "../blueprints/Human.js";



export class Worker extends Human {
    constructor(population_id, name, profession, alive, employed, field_of_work, working, production) {
        super(population_id, name, profession, alive);
        this.employed = employed;
        this.field_of_work = field_of_work;
        this.working = working;
        this.production = production;
    }
}
